hydra-cluster/bench/Bench/EndToEnd.hs

```diff
@@ -218,25 +219,39 @@ movingAverage confirmations =
               )
    in map average fiveSeconds
 
-createUTxOToCommit :: [Dataset] -> FilePath -> IO [UTxO]
-createUTxOToCommit dataset nodeSocket =
-  forM dataset $ \Dataset{fundingTransaction} -> do
-    submit defaultNetworkId nodeSocket fundingTransaction
-    UTxO.min <$> waitForTransaction defaultNetworkId nodeSocket fundingTransaction
+-- | Distribute 100 ADA fuel and starting funds from faucet for each client in
+-- the dataset.
+seedNetwork :: RunningNode -> Dataset -> IO ()
+seedNetwork node@(RunningNode _ nodeSocket) Dataset{fundingTransaction, clientDatasets} = do
+  -- Submit the funding transaction first
+  submit defaultNetworkId nodeSocket fundingTransaction
+  void $ waitForTransaction defaultNetworkId nodeSocket fundingTransaction
+  -- Then, fuel all clients with some 100 ADA
+  forM_ clientDatasets $ \ClientDataset{signingKey} -> do
+    let vk = getVerificationKey signingKey
+    seedFromFaucet defaultNetworkId node vk 100_000_000 Marked
+
+-- | Commit all (expected to exit) 'initialUTxO' from the dataset using the
+-- (asumed same sequence) of clients.
+commitUTxO :: [HydraClient] -> Dataset -> IO UTxO
+commitUTxO clients Dataset{clientDatasets} =
+  mconcat <$> forM (zip clients clientDatasets) doCommit
+ where
+  doCommit (client, ClientDataset{initialUTxO}) = commit client initialUTxO
 
-processTransactions :: [HydraClient] -> [Dataset] -> IO (Map.Map TxId Event)
-processTransactions clients dataset = do
-  let processors = zip (zip dataset (cycle clients)) [1 ..]
+processTransactions :: [HydraClient] -> Dataset -> IO (Map.Map TxId Event)
+processTransactions clients Dataset{clientDatasets} = do
+  let processors = zip (zip clientDatasets (cycle clients)) [1 ..]
   mconcat <$> mapConcurrently (uncurry clientProcessTransactionsSequence) processors
  where
-  clientProcessTransactionsSequence (Dataset{transactionsSequence}, client) clientId = do
-    let numberOfTxs = length transactionsSequence
+  clientProcessTransactionsSequence (ClientDataset{txSequence}, client) clientId = do
```

Comment:
question: why rename the field `transactionsSequence` to `txSequence`? Now the `clientProcessTransactionsSequence` feels a bit odd.

Reply:
That's true. I should have renamed also the function or not rename at all. Why rename? I typed it a couple times during this change and typed it wrong > 50% of those times. The plural `s` just before `Sequence` somehow trips my typing. Also we use the abbreviation `transaction -> tx` pervasively in this code base, so I thought I'll shorten it up.

I will change it back if you insist!?

Reply:
No, I am fine with `tx`  but then perhaps rename the other function which is also a mouthful